service/utils.go

```diff
@@ -21,7 +21,10 @@ const (
 func getEnvironmentId(r *http.Request) (string, error) {
 	environment := r.Header.Get(environmentIdHeader)
 	if environment == "" {
-		return "", fmt.Errorf("Request is missing environment header %s", environment)
+		environment = r.URL.Query().Get("projectId")
```

Comment:
If cattle failed to validate the header, the same would be true. The two are handle the same way in cattle. https://github.com/rancher/cattle/blob/92560ccd0ac372a1a7f632a8053164d14f4c7279/code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/impl/ApiAuthenticator.java#L186-L189